src/main/java/uk/gov/hmcts/reform/cwrdapi/servicebus/TopicPublisher.java

```diff
@@ -1,62 +1,115 @@
 import uk.gov.hmcts.reform.cwrdapi.service.IValidationService;
 import uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 import javax.validation.constraints.NotNull;
 
 @Slf4j
 @Service
 public class TopicPublisher {
 
-    private final JmsTemplate jmsTemplate;
-    private final String destination;
     private final IValidationService validationService;
+    private final MessagingConfig messagingConfig;
 
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Value("${crd.publisher.caseWorkerDataPerMessage}")
+    private int caseWorkerDataPerMessage;
+
+    @Value("${crd.publisher.azure.service.bus.topic}")
+    private String topic;
+
     @Autowired
-    public TopicPublisher(JmsTemplate jmsTemplate,
-                          @Value("${crd.publisher.azure.service.bus.topic}") final String destination,
-                          IValidationService validationService) {
-        this.jmsTemplate = jmsTemplate;
-        this.destination = destination;
+    public TopicPublisher(IValidationService validationService, MessagingConfig messagingConfig) {
         this.validationService = validationService;
+        this.messagingConfig = messagingConfig;
     }
 
-    @Retryable(maxAttempts = 3, backoff = @Backoff(delay = 2000, multiplier = 3))
-    public void sendMessage(@NotNull Object message) {
-        log.info("{}:: Publishing message to service bus topic:: Job Id is: {}", loggingComponentName,
-                validationService.getAuditJobId());
-        if (message instanceof PublishCaseWorkerData) {
+    //The Azure library has retry mechanism inbuilt, which will try to send message until timeout.
+    // There is no need for explicit retry mechanism.
+    public void sendMessage(@NotNull PublishCaseWorkerData caseWorkerIds) {
+        ServiceBusSenderClient serviceBusSenderClient = null;
+        ServiceBusTransactionContext transactionContext = null;
+
+        try {
+            log.info("{}:: Publishing message to service bus topic:: Job Id is: {}", loggingComponentName,
+                    validationService.getAuditJobId());
+
             log.info("{}:: Job Id is: {}, Count of User Ids is: {} ",
-                    loggingComponentName,
-                    validationService.getAuditJobId(),
-                    ((PublishCaseWorkerData) message).getUserIds() != null
-                            ? ((PublishCaseWorkerData) message).getUserIds().size() : null);
-        }
+                    loggingComponentName, validationService.getAuditJobId(), caseWorkerIds.getUserIds().size());
 
-        jmsTemplate.convertAndSend(destination, message);
+            serviceBusSenderClient = messagingConfig.getServiceBusSenderClient();
+            transactionContext = serviceBusSenderClient.createTransaction();
+            publishMessageToTopic(caseWorkerIds, serviceBusSenderClient, transactionContext);
```

Comment:
no need to pass `serviceBusSenderClient ` if you Autowire it

Reply:
Kindly check the above comment.

Reply:
I think yo can Auto-wire all just keep transaction context with little life span if you want to achieve short lived transaction. I think Auto-wiring would simplifies life

Reply:
Hi @SujeethKumarAri @abhijitdiwan I have created the bean for service bus client. Can you please have a look.
Thanks